### PR TITLE
chore(reboot): refresh everything to a like new state but usable

### PR DESCRIPTION
**Pull request by TylorS**
Configure nightwatch to work with TypeScript for E2E tests
Configure webpack and webpack-dev-server for builds, unit tests, intergration tests
Update dependencies to latest version
Implement common npm scripts
Starts fresh, only a counter application is implemented (copy pasta)
Unit tests are there for the firebase authentication driver, this was kept because it was labor-some and definitely required for all features.
No integration tests are currently written
A boilerplate "google nightwatch" e2e test is implemented (copy pasta) to test nightwatch config
An example saucelabs configuration is setup
A lot of needless boilerplate stuff was removed.

### CODE REVIEW

File: .config/nightwatch/sauce.ts [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+    },
+
+    'ie10': {
+      'desiredCapabilities': {
+        'name': 'Windows 8 - IE10',
+        'browserName': 'internet explorer',
+        'platform': 'Windows 8',
+        'version': '10.0',
+      },
+    },
```
Comment (Frikki) [inline .config/nightwatch/sauce.ts:91]: Why IE 9 & 10? IE 9 & 10 have both ~0.3% usage.

File: .config/nightwatch/sauce.ts [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+
+    'safari9': {
+      'desiredCapabilities': {
+        'name': 'Mac 10.11 - Safari 9.0',
+        'browserName': 'safari',
+        'platform': 'OS X 10.11',
+        'version': '9.0',
+      },
+    },
+  },
```
Comment (Frikki) [inline .config/nightwatch/sauce.ts:207]: Safari 7 & 8 have hardly any usage < ~0.3%.

File: custom-typings/nightwatch.d.ts [hunk 3]
```diff
@@ -0,0 +1,1688 @@
...
+    end_session_on_fail: boolean;
+
+    /**
+    * Skip the rest of testcases (if any) when one testcase fails..
+    */
+    skip_testcases_on_fail: boolean;
+
+  }
+
+  export interface TestSuite {
```
Comment (Frikki) [inline custom-typings/nightwatch.d.ts:373]: Why the extra line returns here but not in the interfaces below?

File: dist/index.html [hunk 4]
```diff
@@ -8,7 +8,6 @@
     <meta name="viewport" content="width=device-width, initial-scale=1" />
     <meta name="Content-Langauge" content="en" />
     <link href="https://fonts.googleapis.com/css?family=Open+Sans" rel="stylesheet">
```
Comment (Frikki) [inline dist/index.html]: Remove font loading. It is a no-no for PWAs. Use this template.

``` html
<!doctype html>
<html lang="en">

<head>
  <title>Sparks.Network</title>
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, minimum-scale=1.0, initial-scale=1.0, user-scalable=yes">
  <style>
    body {
      margin: 0;
      font-family: 'Roboto', 'Noto', sans-serif;
      font-size: 13px;
      line-height: 1.5;
      min-height: 100vh; }
  </style>
</head>

<body>

  <div id="sparks-app" class="unresolved">
    <div>sparks<span>.network<span></div>
  </div>

  <script>window.performance && performance.mark && performance.mark('index.html')</script>
  <script src="/bundle.js"></script>
  <!--
    <script>
      !function(f,b,e,v,n,t,s){if(f.fbq)return;n=f.fbq=function(){n.callMethod?
      n.callMethod.apply(n,arguments):n.queue.push(arguments)};if(!f._fbq)f._fbq=n;
      n.push=n;n.loaded=!0;n.version='2.0';n.queue=[];t=b.createElement(e);t.async=!0;
      t.src=v;s=b.getElementsByTagName(e)[0];s.parentNode.insertBefore(t,s)}(window,
      document,'script','https://connect.facebook.net/en_US/fbevents.js');
      fbq('init', '1660062114278795');
      fbq('track', "PageView");
    </script>
    <noscript><img height="1" width="1" style="display:none" src="https://www.facebook.com/tr?id=1660062114278795&ev=PageView&noscript=1"/></noscript>
    -->

</body>
</html>
```

File: src/app.ts [hunk 5]
```diff
@@ -1,40 +1,50 @@
...
-  random: () => just(Math.random())
-};
-
-Cycle.run(main, drivers);
+import { Stream, merge, constant, scan, map } from 'most';
+import { run, DriverFn } from '@motorcycle/core';
+import { makeDOMDriver, DOMSource, VNode, div, h2, button } from '@motorcycle/dom';
+
+export interface MainSources {
+  DOM: DOMSource;
```
Comment (Frikki) [inline src/app.ts]: I would prefer to see this lowercased, it is not a CONSTANT.

**Review by Frikki — CHANGES_REQUESTED** [review]
Minor comments and requests.

File: .config/nightwatch/sauce.ts (hunk 1 above)
Comment (TylorS) [reply .config/nightwatch/sauce.ts:91]: They were copy and paste

File: .config/nightwatch/sauce.ts (hunk 2 above)
Comment (TylorS) [reply .config/nightwatch/sauce.ts:207]: They were copy and paste

File: custom-typings/nightwatch.d.ts (hunk 3 above)
Comment (TylorS) [reply custom-typings/nightwatch.d.ts:373]: This nightwatch typings were copied from a gist

**Review by Frikki — APPROVED** [review]

**Pull request merged**